src/redux/Auth/auth.js

```diff
@@ -0,0 +1,68 @@
+import { authUser, regUser } from '../../helpers/authUser';
+
+const REGISTER_USER = 'auth/user/REGISTER_USER';
+const SIGN_IN_USER = 'auth/user/SIGN_IN_USER';
+const SIGN_OUT_USER = 'auth/user/SIGN_OUT_USER';
+const GET_ERRORS = 'auth/user/GET_ERRORS';
+
+const initialState = {
+  logged_in: false,
+  loading: true,
+  token: null,
+  details: null,
+  errors: null,
+};
+
+export const createUser = (data) => async (dispatch) => {
+  const res = await regUser(data);
+  if (res.token) {
+    dispatch({ type: REGISTER_USER, payload: res });
+  } else {
+    dispatch({ type: GET_ERRORS, payload: res.errors[1] });
+  }
+};
+
+export const logOutUser = () => {
+  localStorage.removeItem('token');
+
+  return { type: SIGN_OUT_USER };
+};
+
+export const logInUser = (data) => async (dispatch) => {
+  const res = await authUser(data);
+  if (res.token) {
+    dispatch({ type: SIGN_IN_USER, payload: res });
+  } else {
+    dispatch({ type: GET_ERRORS, payload: res.errors });
+  }
+};
+
+const authReducer = (state = initialState, action) => {
+  const { type, payload } = action;
+  switch (type) {
+    case REGISTER_USER:
+      return {
+        ...state, logged_in: true, loading: false, token: payload.token, details: payload.details,
+      };
+    case GET_ERRORS:
+      return {
+        ...state, logged_in: false, loading: false, errors: payload,
+      };
+    case SIGN_IN_USER:
+      return {
+        ...state,
+        logged_in: true,
+        loading: false,
+        token: payload.token,
+        details: payload.details,
+        errors: null,
+      };
```

Comment:
[Optional]
Same here. You can add admin: payload.admin.

Reply:
I've added it